### PR TITLE
https://issues.redhat.com/browse/ACM-14756 Add API doc

### PR DESCRIPTION
**Pull request by oafischer**
Train 19

### CODE REVIEW

File: clusters/api/klusterletconfig.json.adoc [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+* klusterletconfigs.config.open-cluster-management.io : Create and manage klusterletconfigs
+
+
+[[_rhacm-docs_apis_klusterletconfig_jsonpaths]]
+== Paths
+
+[[_rhacm-docs_apis_klusterletconfig_jsonqueryklusterletconfigs]]
+=== Query a single KlusterletConfig 
+....
+GET /cluster.open-cluster-management.io/v1beta1/namespaces/{namespace}/klusterletconfigs
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc]:
```suggestion
GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name}
```

File: clusters/api/klusterletconfig.json.adoc [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+    },
+    "conditions": [],
+    "storedVersions": []
+  }
+}
+----
+
+
+[[_rhacm-docs_apis_klusterletconfig_jsonqueryklusterletconfig]]
+=== Query a single klusterletconfig
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc:456]: this should not be the same as https://github.com/stolostron/rhacm-docs/pull/7186/files#diff-2340f531322fc2927b5f523f4bfd962aea3e6928d762084f27537cdd49faa2dfR27

File: clusters/api/klusterletconfig.json.adoc [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+....
+GET /klusterletconfigs.config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name}
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc]:
```suggestion
GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name}
```

File: clusters/api/klusterletconfig.json.adoc [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+
+==== Tags
+
+* klusterletconfigs.config.open-cluster-management.io
+
+
+[[_rhacm-docs_apis_klusterletconfig_jsondeleteklusterletconfig]]
+=== Delete a klusterletconfig
+....
+DELETE /klusterletconfigs.config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name}
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc]:
```suggestion
DELETE /config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name} 
```

File: clusters/api/klusterletconfig.json.adoc [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+|*403*|Access forbidden|No Content
+|*404*|Resource not found|No Content
+|*500*|Internal service error|No Content
+|*503*|Service unavailable|No Content
+|===
+
+
+==== Tags
+
+* managedserviceaccounts.authentication.open-cluster-management.io
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc]: it is not managedserviceaccounts

File: clusters/api/klusterletconfig.json.adoc [hunk 6]
```diff
@@ -0,0 +1,558 @@
+[#klusterletconfig-api]
+= KlusterletConfig API (v1alpha1)
+
+
+[[_rhacm-docs_apis_klusterletconfig_jsonoverview]]
+== Overview
+
+This documentation is for the `KlusterletConfig` resource for the {mce}. The `KlusterletConfig` resource is used to configure the Klusterlet installation. The four possible requests are: create, query, delete, and update.
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc:8]: I did not find description for `update`

Comment (oafischer) [reply clusters/api/klusterletconfig.json.adoc:8]: There is not supposed to be a description for `update`, this is the same for all of our APIs.

**Comment by oafischer** [comment]
Made some updates, let me know if it looks good now @zhujian7. Thanks!

**Comment by zhujian7** [comment]
@oafischer there are 2 
```
=== Query a single KlusterletConfig
GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name}
```
line 27-34 and line 459-466, I am not sure if line 27-34 should be changed to something like `Query all KlusterletConfigs`?

**Comment by oafischer** [comment]
@zhujian7 We use "query a single API" for all our GET sections in the API doc. If that is not accurate or it would be better to say "all", we can discuss that in a separate issue and maybe update all of the API doc at the same time. I'd prefer to stick to just klusterconfig stuff in this PR.

**Comment by zhujian7** [comment]
> @zhujian7 We use "query a single API" for all our GET sections in the API doc. If that is not accurate or it would be better to say "all", we can discuss that in a separate issue and maybe update all of the API doc at the same time. I'd prefer to stick to just klusterconfig stuff in this PR.

@oafischer I still do not understand why we need two "Get a single klusterletconfig", this does not make sense to me:

[line 26-34](https://github.com/stolostron/rhacm-docs/pull/7186/files#diff-2340f531322fc2927b5f523f4bfd962aea3e6928d762084f27537cdd49faa2dfR23-R34)
```
[[_rhacm-docs_apis_klusterletconfig_jsonqueryklusterletconfigs]]
=== Query a single KlusterletConfig 
....
GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name}
....


==== Description
Query a single `KlusterletConfig` for more details.
```

[line 458-466](https://github.com/stolostron/rhacm-docs/pull/7186/files#diff-2340f531322fc2927b5f523f4bfd962aea3e6928d762084f27537cdd49faa2dfR458-R466)
```
[[_rhacm-docs_apis_klusterletconfig_jsonqueryklusterletconfig]]
=== Query a single klusterletconfig
....
GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name}
....



==== Description
Query a single `KlusterletConfig` for more details.
```

**Comment by zhujian7** [comment]
BTW, we do have Query all xxx, for example, 
Query all clusters: https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.11/html/clusters/cluster_mce_overview#rhacm-docs_apis_cluster_jsonqueryclusters
Query all clustersets: https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.11/html/clusters/cluster_mce_overview#rhacm-docs_apis_clusterset_jsonqueryclustersets

**Comment by oafischer** [comment]
@zhujian7 OK I will change it to Query all and remove the second Query.

**Comment by zhujian7** [comment]
> @zhujian7 OK I will change it to Query all and remove the second Query.

@oafischer  no, compared to other APIs, I do not think we should remove the second Query, 
- this first one should be:
```
[[_rhacm-docs_apis_klusterletconfig_jsonqueryklusterletconfigs]]
=== Query all klusterletconfigs
....
GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs # without the "/{klusterletconfig_name}" ending
....


==== Description
Query your `KlusterletConfigs` for more details
```

- the second one should be kept as no change:
```
[[_rhacm-docs_apis_klusterletconfig_jsonqueryklusterletconfig]]
=== Query a single klusterletconfig
....
GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name} # with the "/{klusterletconfig_name}" ending
....



==== Description
Query a single `KlusterletConfig` for more details.
```

**Comment by oafischer** [comment]
I see, I'll add it back

**Comment by zhujian7** [comment]
@oafischer the cluster API, there are [Query all clusters](https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.11/html/clusters/cluster_mce_overview#rhacm-docs_apis_cluster_jsonqueryclusters) and [Query a single cluster](https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.11/html/clusters/cluster_mce_overview#rhacm-docs_apis_cluster_jsonquerycluster)

File: clusters/api/klusterletconfig.json.adoc [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+* klusterletconfigs.config.open-cluster-management.io : Create and manage klusterletconfigs
+
+
+[[_rhacm-docs_apis_klusterletconfig_jsonpaths]]
+== Paths
+
+[[_rhacm-docs_apis_klusterletconfig_jsonqueryklusterletconfigs]]
+=== Query all KlusterletConfig 
+....
+GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs/{klusterletconfig_name}
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc]:
```suggestion
GET /config.open-cluster-management.io/v1alpha1/klusterletconfigs
```

File: clusters/api/klusterletconfig.json.adoc [hunk 8]
```diff
@@ -0,0 +1,556 @@
...
+
+
+==== Parameters
+
+[options="header", cols=".^2a,.^3a,.^9a,.^4a"]
+|===
+|Type|Name|Description|Schema
+|*Header*|*COOKIE* +
+__required__|Authorization: Bearer {ACCESS_TOKEN} ; ACCESS_TOKEN is the user access token.|string
+|*Path*|*klusterletconfig_name* +
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc]:
```suggestion
```

File: clusters/api/klusterletconfig.json.adoc [hunk 9]
```diff
@@ -0,0 +1,556 @@
...
+__required__|Name of the `KlusterletConfig` that you want to query.|string
```
Comment (zhujian7) [inline clusters/api/klusterletconfig.json.adoc]:
```suggestion
```

**Comment by oafischer** [comment]
Are any more changes needed? @zhujian7

File: clusters/api/klusterletconfig.json.adoc [hunk 10]
```diff
@@ -0,0 +1,554 @@
...
+|Name|Description|Schema
+|*apiVersion* +
+__required__|The versioned schema of the klusterletconfig. |string
+|*kind* +
+__required__|String value that represents the REST resource. |string
+|*metadata* +
+__required__|The meta data of the `KlusterletConfig`. |object
+|*spec* +
+__required__|The specification of the `KlusterletConfig`. |
+|===
```
Comment (dockerymick) [inline clusters/api/klusterletconfig.json.adoc:554]: My only concern in this file is the future tense used in the description of the Request body. I assume this is copied from the example that is provided to you. We should consider helping to improve these descriptions when we have the bandwidth. I see that "will" exists in the request body for the managed_serviceaccount.json.adoc and multicluster_engine.json.adoc. i also noticed that Version information is not mentioned in any of the APIs of the clusters folder. I didn't know when were doing that, but at least the APIs in the `clusters` folder are consistent

**Review by dockerymick — APPROVED** [review]

File: clusters/api/klusterletconfig.json.adoc (hunk 10 above)
Comment (oafischer) [reply clusters/api/klusterletconfig.json.adoc:554]: Agreed, I don't think it's worth editing all the descriptions on our end if the plan is to pull them from the source in the future anyway. We can discuss another round of API edits for next release like we did a while back.

All the API doc needs an overhaul anyway, I'll be working on that next release as well.

**Pull request merged**